### PR TITLE
Case tile template selection backend

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

We're adding support for case tile templates. [Here](https://dimagi-dev.atlassian.net/browse/USH-3160?atlOrigin=eyJpIjoiM2RlZTJiMzdkYWI0NDg2ZmI1MjVjOTkzMDhmZDE5N2EiLCJwIjoiaiJ9) is the work in Jira and [here](https://docs.google.com/document/d/1DqeTK-X8Z8sdwyQjXLGFmcwCFVq1gt0cm_9MLzTIlwo/edit#heading=h.d7s2sycdfk92) is the spec.

This PR covers the backend code and sets us up for [UI work](https://dimagi-dev.atlassian.net/browse/USH-3161) to be done next.

## Technical Summary

This PR adds the backend code with a wrap function to transition us from `Detail.use_case_tiles` (boolean) `Detail.case_tile_template` (string).

Originally we planned to do this in a single migration with deploy, but it turned out more apps have case tiles than we thought, and the migration might be too big for that. So this PR utilizes a wrap function to allow us to run a migration between deploys (we're planning on running between next Wed's deploy and the deploy after).

PR #33002 has the actual django migration and migration command, which we're planning on deploying  in two weeks, 6/7).

I recommend reviewing by commit due to the cumbersome app `.json` files!

## Feature Flag

CASE_LIST_TILE

### Safety story

Tested locally and on staging.

### Automated test coverage

Existing tests cover case tiles.

### QA Plan

Not for this PR.

### Migrations

This PR does not include migrations, but sets up one that can be found in #33002.

### Rollback instructions

- [] This PR can be reverted after deploy with no further considerations

If this code is run on an environment then some apps will have the new couch data due to the wrap function and the code in this PR will be required to support those apps.

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by MartinRiese — CHANGES_REQUESTED** [review]
I just checked out your branch and opened an app that I knew that "Use case tiles" enabled. But it now showed "Don't use case tiles". I am not sure if a migration is needed but for existing app that use tile somehow the value needs to be set correctly.

**Review by MartinRiese — APPROVED** [review]

**Comment by nospame** [comment]
> So I was able to reproduce the bug Martin described once, but since then Martin and I have both had trouble recreating it, even if we create entirely new apps.
> 
> So, reviewers please let me know if you have any ideas!

I was able to recreate this once by having a changed and saved, but not built/published change on an app that uses case tiles, then switching to your branch. I tried again with a different case tiles app and could not recreate it the same way. So, no super great leads for you.

**Review by snopoke — APPROVED** [review]

File: corehq/apps/app_manager/static/app_manager/js/details/screen.js [hunk 1]
```diff
@@ -56,9 +56,9 @@ hqDefine("app_manager/js/details/screen", function () {
         self.containsCustomXMLConfiguration = options.containsCustomXMLConfiguration;
         self.allowsTabs = options.allowsTabs;
         self.caseTileFields = options.caseTileFields;
-        self.useCaseTiles = ko.observable(spec[self.columnKey].use_case_tiles ? "yes" : "no");
+        self.caseTileTemplate = ko.observable(spec[self.columnKey].case_tile_template ? "yes" : "no");
         self.showCaseTileColumn = ko.computed(function () {
-            return self.useCaseTiles() === "yes" && hqImport('hqwebapp/js/toggles').toggleEnabled('CASE_LIST_TILE');
+            return self.caseTileTemplate() === "yes" && hqImport('hqwebapp/js/toggles').toggleEnabled('CASE_LIST_TILE');
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/details/screen.js:61]: Isn't `caseTileTemplate` going to be `person_simple` initially?

**Review by orangejenny — COMMENTED** [review]
Left one comment that might be a bug. Even if it isn't, I think it'd be simpler to get rid of the yes/no logic entirely at this point.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**